api/test/plugin/dynamic_load_test.cc

```diff
@@ -2,6 +2,12 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "opentelemetry/plugin/dynamic_load.h"
+#ifdef _WIN32
```

Comment:
Why is this needed as the `#ifdef _WIN32` is included in `opentelemetry/plugin/dynamic_load.h` which has already been included above?

Reply:
In `dynamic_load.h` file, there is no preprocessor definition anymore, I removed it in this pull request. Because there is only one function declaration which is not related to operating system itself. 

However, only this source file is related to operating system itself with dynamic_load and which has to be included like that. Otherwise, there will be a compiler error on test codes.

Reply:
Why was the including of such `dynamic_load_<os>.h` moved out of `dynamic_load.h`? With change, the user code like the test code here need to include both files separately. It looks to me that including the os header in `dynamic_load.h` is more convenient.

Reply:
Ok, I will revert it back.